File: adt2extn/_patts_marked.c
// This file is part of Adlib Tracker II (AT2).
//
// SPDX-FileType: SOURCE
// SPDX-FileCopyrightText: 2014-2025 The Adlib Tracker 2 Authors
// SPDX-License-Identifier: GPL-3.0-or-later

uint8_t _patts_marked (void)
{
  uint8_t result = 0;

  for (uint8_t i = 0; i < 0x80; i++)
    if (songdata.pattern_names[i][1] == charmap.black_right_triangle)
      result++;

  return result;
}

File: adt2extn/pas/_patts_marked.pas
// This file is part of Adlib Tracker II (AT2).
//
// SPDX-FileType: SOURCE
// SPDX-FileCopyrightText: 2014-2025 The Adlib Tracker 2 Authors
// SPDX-License-Identifier: GPL-3.0-or-later

function _patts_marked: Byte;

var
  temp,
  result: Byte;

begin
  result := 0;
  For temp := 0 to $7f do
    If (songdata.pattern_names[temp][1] = #16) then
      Inc(result);
  _patts_marked := result;
end;
